tests/test_compile_kwargs.py

```diff
@@ -0,0 +1,555 @@
+        assert compiled_metrics[metric_idx + 1].name == "out1_name_all_metrics"
+        assert compiled_metrics[metric_idx + 2].name == "out2_custom_name"
+        assert compiled_metrics[metric_idx + 3].name == "out2_name_all_metrics"
+
+
+def test_metrics_routed_params_dict():
+    """Tests compiling metrics with routed parameters
+    when they are passed as a dict.
+    """
+    n_outputs_ = 2
+
+    metrics = metrics_module.BinaryAccuracy
+
+    X, y = make_classification()
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    # loss functions for each output and joined show up as metrics
+    metric_idx = 1 + n_outputs_
+
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss="binary_crossentropy",
+        metrics={"out1": metrics},
+        metrics__out1__name="custom_name",
+    )
+    est.fit(X, y)
+    assert est.model_.metrics[metric_idx].name == "out1_custom_name"
+
+    if n_outputs_ == 1:
+        metrics_ = ({"out1": metrics},)
+    else:
+        metrics_ = {f"out{i+1}": metrics for i in range(n_outputs_)}
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss="binary_crossentropy",
+        metrics=metrics_,
+        metrics__name="name_all_metrics",  # ends up out2 only
+        metrics__out1__name="custom_name",  # ends up in out1 only
+    )
+    est.fit(X, y)
+    assert est.model_.metrics[metric_idx].name == "out1_custom_name"
+    assert est.model_.metrics[metric_idx + 1].name == "out2_name_all_metrics"
+
+
+def test_metrics_invalid_string():
+    """Tests that a ValueError is raised when an unknown
+    string is passed as a metric.
+    """
+
+    X, y = make_classification()
+
+    metrics = [
+        "acccuracy",
+    ]  # acccuracy (extra `c`) is not a metric
+
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss="binary_crossentropy",
+        metrics=metrics,
+    )
+    with pytest.raises(ValueError, match="Unknown metric function"):
+        est.fit(X, y)
+
+
+def test_metrics_uncompilable():
+    """Tests that a TypeError is raised when a metric
+    that is not compilable is passed routed parameters.
+    """
+
+    X, y = make_classification()
+
+    metrics = [
+        metrics_module.get("accuracy"),
+    ]  # a function
+
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss="binary_crossentropy",
+        metrics=metrics,
+        metrics__name="custom_name",
+    )
+    expected_param = {"name": "custom_name"}
+    with pytest.raises(
+        TypeError,
+        match=f'TypeError: "{str(metrics[0])}" object of type "{type(metrics[0])}"'
+        "does not accept parameters because it's not a class."
+        f' However, it received parameters "{expected_param}"',
```

Comment:
I'd only keep the core part of this message, `"does not accept parameters because it's not a class"`.

Reply:
Sure, does make it cleaner.